File: BackEnd/Models/Playlist.cs
using System;
using System.Collections.Generic;

namespace Showreel.Models;

public partial class Playlist
{
    public int Id { get; set; }

    public string? Status { get; set; }

    public DateTime? StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public string? Title { get; set; }

    public string? JsonPlaylist { get; set; }

    public string? JsonListVideo { get; set; }

    public string? Creator { get; set; }

    public int? ParentId { get; set; }
}

File: FrontEnd/src/model/playlist.ts
type statusPlaylist = 'pending' | 'active' | 'expired';

interface IBuildingDetail {
    id: number;
    buildingName: string;
}
export interface IPostPlaylistStore {
    id: number;
    status: statusPlaylist;
    startDate?: string;
    endDate?: string;
    title: string;
    jsonPlaylist: string;
    JsonListVideo?: string;
    creator: string;
    parentId: number;
    buildingsId?: number[];
    buildings?: IBuildingDetail[];
}
